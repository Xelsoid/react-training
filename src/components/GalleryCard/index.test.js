import React from 'react';
import MOCKED_DATA from '@mockedData/MOCKED_DATA';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ComponentExample from './index';

const mockStore = configureStore([]);

describe('GalleryCard test', () => {
  let store;
  const film = MOCKED_DATA.data[1];

  beforeEach(() => {
    store = mockStore({
      movieData: film,
    });
  });

  it('renders correctly', () => {
    const component = shallow(
      <Provider store={store}>
        <ComponentExample {...film} />
      </Provider>,

    );
    expect(component).toMatchSnapshot();
  });
});
