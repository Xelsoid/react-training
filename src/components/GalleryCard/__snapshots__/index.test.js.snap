// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryCard test renders correctly 1`] = `
<div
  className="gallery-card"
>
  <img
    alt="Star Wars: The Last Jedi"
    className="gallery-card__image"
    src="https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg"
  />
  <div
    className="gallery-card__description-line"
  >
    <span
      className="gallery-card__name"
    >
      Star Wars: The Last Jedi
    </span>
    <span
      className="gallery-card__year"
    >
      2017
    </span>
  </div>
  <small
    className="gallery-card__genre"
  >
    Fantasy 
    Adventure 
    Science Fiction 
  </small>
</div>
`;
